code/modules/antagonists/voidwalker/sunwalker_mob.dm

```diff
@@ -0,0 +1,81 @@
+/// Voidwalker murderbone variant focused around the sun (wow yet another fire element antag!!)
+/mob/living/basic/voidwalker/sunwalker
+	name = "Heliarch"
+	desc = "A cosmic entity covered in stellar corona. You DEFINITELY shouldn't stare."
+
+	icon_state = "sunwalker"
+
+	melee_damage_type = BURN
+	melee_damage_lower = 25
+	melee_damage_upper = 25
+
+	health = 200
+	maxHealth = 200
+
+	obj_damage = 50
+	speed = 0.2
+
+	maximum_survivable_temperature = INFINITY
+	pressure_resistance = INFINITY
+
+	charge = /datum/action/cooldown/mob_cooldown/charge/sunwalker
+	telepathy = /datum/action/cooldown/spell/list_target/telepathy/voidwalker/sunwalker
+	can_do_abductions = FALSE
+
+	regenerate_colour = COLOR_BLUE
+	light_range = 7
+
+	/// Temperature we move our surroundings towards
+	var/hotspot_temperature = 1000
+	/// Gas volume passively exposed to our temperature
+	var/hotspot_volume = 100
+
+	/// Water damage we take on any exposure
+	var/water_damage = 10
+	/// Below this health threshold, we dont take water damage
+	var/water_damage_cutoff = 10
+
+/mob/living/basic/voidwalker/sunwalker/unique_setup()
+	AddComponent(/datum/component/igniter)
+	AddComponent(/datum/component/vision_hurting, damage_per_second = 0.1, message = null, silent = TRUE)
+	AddComponent(/datum/component/space_dive, /obj/effect/dummy/phased_mob/space_dive/sunwalker)
+
+	create_reagents(1) // Needed for the water reagent interactions to work
+
+	var/static/list/ray_filter = list(type = "rays", size = 32, color = COLOR_BLUE_VERY_LIGHT)
+	add_filter("sunwalker_rays", 3, ray_filter)
+	animate(get_filter("sunwalker_rays"), offset = 1, time = 3 SECONDS, loop = -1, flags = ANIMATION_PARALLEL)
+
+/mob/living/basic/voidwalker/sunwalker/examine(mob/user)
+	. = ..()
+
+	if(!iscarbon(user))
```

Comment:
It's a shame this only applies to carbons but I guess it's not worth inventing traits purely to decide whether a non-human mob sees using eyes or not